### ISSUE TITLE
Use Weblate for translations

### ISSUE THREAD

**Issue by ignotus666** [jamulussoftware/jamuluswebsite#762]
@jamulussoftware/translators @ann0see @gilgongo 

Though the po4a setup made things a bit simpler, I think there are still big barriers when it comes to translating the documentation. A big one is having to navigate all the GitHub stuff (forking, branches, PRs, conflicts, etc.). Further problems arise with local translating due to different OS's, editors/versions, incorrect settings, etc. A case in point is the ongoing German translation, where this poor person had to jump through countless hoops and waste a lot of time.

I had another look at [Weblate](https://weblate.org/en-gb/). Back in the day when I checked it out I found it quite daunting and also failed to see that there is an option of free hosting for libre projects (you otherwise need to either pay or host the service yourself). I've set up a trial account linked to my repo and have been doing some testing, and it looks quite promising - basically what it does is shift the translation process itself to an online interface. Translators only need a GitHub account and don't need to worry about forks, PRs or any of that, as Weblate itself creates forks and you can configure it to automatically push commits, create PRs, or push manually. It also pulls changes to keep in sync with the repo. There are nice features such as highlighting source file changes, checking for spacing, line break and other errors, displaying translation progress charts on your repo, and a lot of other stuff I still need to get my head around. Incidentally, the po4a project itself uses Weblate.
Translators just have to log in to Weblate, click through to the documents they want to translate and do their translations there, the rest is taken care of, either automatically, or by admins. The po4a setup and all the GH actions on our repo would stay as they are.

I reckon it'd a bit premature to roll it out for this release and first I wanted to see what you think. The project needs to qualify for free hosting, but I think we meet the requirements (you can see them [here](https://hosted.weblate.org/hosting/)).

You can check out the trial version I set up [here](https://hosted.weblate.org/projects/ignotus/).

**Comment by ignotus666** [comment]
I just realised this might be better in Discussions... feel free to move it.

**Comment by ann0see** [comment]
Hmm yes after looking at the page it looks quite promising. I'd however prefer not needing to have another platform to maintain

**Comment by ignotus666** [comment]
True, but conversely it could save a lot of hand-holding and make the translation process much more accessible by removing the requirement of needing to know how to use GitHub.

**Comment by BLumia** [comment]
> True, but conversely it could save a lot of hand-holding and make the translation process much more accessible by removing the requirement of needing to know how to use GitHub.

Additionally, Weblate also send translation updates via GitHub Pull Request automatically, so if choosing Weblate, we can still accept regular GitHub Pull Request if a translator doesn't want to use the hosted Weblate instance.

One thing might be worth to mention is, if we choose to use Weblate, we'll need to pay more attention to the source strings that contains arguments (e.g. `"About %1"`) and make use of the 2nd `disambiguation` argument in `QObject::tr()` to tell translator what's the argument means, since unlike using QtLinguist with full source code fetched, the translation platform won't show the related source code in the translation page.

I'm using Weblate for my tiny side project and so far it's pretty good for me. You can check out [here](https://hosted.weblate.org/projects/pineapple-pictures/application/) for what it looks like for the Weblate translation process (how to add a new language, how to translate a string, etc.), and [here](https://github.com/BLumia/pineapple-pictures/pull/54) for a Pull Request that automatically sent by Weblate to update the ts files.

HTH

**Comment by ann0see** [comment]
Yeah. Valid point. I'd say it's up to @jujudusud @henkdegroot too if they want to use it. I used transifex some time ago (or however it's called) and it was convenient.

**Comment by henkdegroot** [comment]
The current setup works fine for me. Not against using anything else, however when I do translation(s) I usually want to be able to "see" the end result before submitting a PR. Hence I already need to install/download code etc. 

Having a less github involved setup could benefit to attract new translators.

**Comment by trebmuh** [comment]
Hi all. I'm using Weblate for a unch of other project I'm translating and I :+1: this proposition. Weblate is very convenient for translators as it removes all the "non-translation" barrier (code,...). One can use github to connect to it or use a login/password if he doesn't have a github account. This is a good thing and a bad thing in the same time and I recommand you to set up a "reviewer" for language with more than 1 translator. Unless that you're at risk to have editing conflicts between people.

**Comment by ann0see** [comment]
Ok. So I'm not going to stop you from implementing it. However no guarantee for being online there.
